### PR TITLE
GBA: Verify ELF entrypoint against ROM header

### PR DESCRIPTION
**Pull request by LunarLambda**
Detect more ELF binaries as valid by checking if the ROM/Multiboot entry branches in the header branch to `e_entry`.

We do this by locating the program header containing the entry branch address, decoding it, checking if it's a branch or branch-with-link instruction, and checking if the branch's target address is `e_entry`.

Should fix detection of Multiboot ROMs built with devkitARM and gba-toolchain.

### CODE REVIEW

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:618]: You can reduce the indent level by inverting this check and making it a continue instead.

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:620]:
```suggestion
			const char* bytes = ELFBytes(elf, &esize);
```

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:623]: The parens here don't affect the result and don't help with readability.

```suggestion
			size_t off = phdr->p_offset + target - phdr_s;
```

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:634]: See above about reducing the indent level.

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:635]: See above about paren usage.
```suggestion
				uint32_t b_target = target + info.op1.immediate + 8;
```

**Review by endrift — CHANGES_REQUESTED** [review]
Should be good to merge after some nit cleanup. Would probably be a good idea to add a CHANGES entry too (under the misc section)

**Comment by LunarLambda** [comment]
:+1:

File: CHANGES
Comment (endrift) [inline CHANGES:18]: Please keep these sections alphabetized by category.

File: include/mgba/internal/gba/gba.h
Comment (endrift) [inline include/mgba/internal/gba/gba.h:59]: This belongs in the ifdef below, right above the function declaration.

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:618]:
```suggestion
		if (target < phdr_s || target + 4 > phdr_e) {
```
That sure was a bunch of parens. Same reasoning as the previous nit cleanup; sorry I missed it that time.

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:637]:
```suggestion
		if (info.branchType != ARM_BRANCH && info.branchType != ARM_BRANCH_LINKED) {
```
Same logic as above.

**Review by endrift — CHANGES_REQUESTED** [review]
Some minor corrections to the last pass, as well as two paren nits I missed before. There shouldn't be any nits after this since I did a full pass to make sure I didn't miss any other paren style nits.

Generally speaking, variable names in mGBA are `camelCase` not `snake_case`, but I won't hold up this PR on naming those given these specific names (only one letter on one of the two "words"), but if you wanna fix those too I won't complain.

File: src/gba/gba.c
Comment (endrift) [inline src/gba/gba.c:633]:
```suggestion
		LOAD_32(opcode, off, bytes);
```
That's what that middle field is for :P

That said, an additional alignment check is a good idea, as unaligned 4-byte loads are unspecified on ARM7TDMI (and the way it actually works is weird and not worth emulating here--we can just reject an unaligned address out of hand since that would break in actual use(

**Review by endrift — APPROVED** [review]

**Pull request merged**
